listenbrainz/listenstore/timescale_listenstore.py

```diff
@@ -235,17 +235,23 @@ def fetch_listens_from_storage(self, user_name, from_ts, to_ts, limit, order, ti
             max_timestamp_window = -1
         else:
             max_timestamp_window = 432000 * time_range
-            if from_ts is not None:
+            if from_ts and to_ts:
+                pass
+            elif from_ts is not None:
                 to_ts = from_ts + max_timestamp_window
-            else:
+            elif to_ts is not None:
                 from_ts = to_ts - max_timestamp_window
-
+                
         query = """SELECT listened_at, track_name, created, data
                      FROM listen
                     WHERE user_name = :user_name """
+                  
 
         if max_timestamp_window < 0:
-            if from_ts is not None:
+            if from_ts and to_ts:
+                query += """AND listened_at > :from_ts
+                            AND listened_at < :to_ts """
```

Comment:
We talked recently about limiting the maximum duration that you can query (to about a year I think?). Does the web API or this method perform any checking of this limit?

Reply:
The API performs this check -- this limit should not exist on the listenstore level, since that would preclude internal processes from accessing larger time ranges.

Reply:
This will have to be rebased to pull the new range checking code in - however, the current code only checks if the `time_range` parameter is less than a set maximum:  https://github.com/metabrainz/listenbrainz-server/blame/c9067520546a77146e992f8e08a4647a9d1c5e1d/listenbrainz/webserver/views/api.py#L107

this PR will also need to check in the API endpoint that the time between from_ts and to_ts is less than this range.